internal/discord/handler/handler.go

```diff
@@ -0,0 +1,43 @@
+package handler
+
+import (
+	"github.com/bwmarrin/discordgo"
+	"github.com/seventv/common/utils"
+	"go.uber.org/zap"
+)
+
+var (
+	DefaultRoleId = ""
+)
+
+func Register(session *discordgo.Session) {
+	if DefaultRoleId == "" {
+		zap.S().Fatalw("DefaultRoleId is not set")
+
+		return
+	}
+
+	session.AddHandler(messageCreate)
+	session.AddHandler(guildMemberAdd)
```

Comment:
In order to pass the global context to these handlers, wrap them in another function, so you end up with a signature like this

```go
func messageCreate(gctx global.Context) func (s *discordgo.Session, m *discordgo.MessageCreate) {
    return func(...)
}
```

then call `messageCreate()` as the argument to `session.AddHandler()`